paddle/fluid/pybind/eager_method.cc

```diff
@@ -1444,6 +1445,36 @@ static PyObject* tensor__copy_gradient_from(TensorObject* self,
   EAGER_CATCH_AND_THROW_RETURN_NULL
 }
 
+static PyObject* tensor__use_cudnn(TensorObject* self,
+                                   PyObject* args,
+                                   PyObject* kwargs) {
+  EAGER_TRY
+  PADDLE_ENFORCE(self->tensor.defined() && self->tensor.is_dense_tensor(),
+                 paddle::platform::errors::Fatal(
+                     "function _use_cudnn is only effective for DenseTensor"));
+
+  bool use_cudnn = pybind::CastPyArg2AttrBoolean(PyTuple_GET_ITEM(args, 0), 0);
+
+  phi::DenseTensor* dense_tensor =
+      static_cast<phi::DenseTensor*>(self->tensor.impl().get());
+  phi::DenseTensorMeta target_dense_meta =
+      *(phi::DenseTensorUtils::GetMutableMeta(dense_tensor));
+  target_dense_meta.use_cudnn = use_cudnn;
+  // Share all other members of Tensor except use_cudnn
+  phi::DenseTensor target_dense_tensor;
+  target_dense_tensor.ShareDataWith(*dense_tensor);
+  target_dense_tensor.set_meta(target_dense_meta);
+  paddle::experimental::Tensor target_tensor(
+      std::make_shared<phi::DenseTensor>(target_dense_tensor),
+      self->tensor.name());
+  target_tensor.set_autograd_meta(self->tensor.mutable_autograd_meta());
```

Comment:
为什么需要创建一个新的Tensor？

Reply:
参考前面 review 意见的回复：

> 此函数需要返回 tensor 对象，返回的 tensor 与原 tensor 相比，只有 use_cudnn 属性不一致。
一个 tensor 可以同时作为多个 op 的输入，如果对 tensor 使用 inplace 的 set 方法，会污染其他 op 的行为，因此需要返回一个新的 tensor。
例如 Tensor x 同时作为 conv2d 和 depthwise_conv2d 的输入，如果只有 depthwise_conv2d 想要使用 plain GPU kernel，直接 set x.use_cudnn = False 时，conv2d 也会选取 plain GPU kernel。因此需要返回一个新的 tensor 隔离 op 之间选择 kernel 的方式。

Referring to the above review comments:

> This function has to return a tensor object, the returned tensor is the same as the original tensor except use_cudnn attribute.
A tensor can be the input of multiple ops at the same time. It will contaminate other ops' action if we set use_cudnn attribute inplace, thus we need to return a new tensor.
For example, Tensor x is the input of both conv2d and depthwise_conv2d. However, only depthwise_conv2d wants to use plain GPU kernel and conv2d intends to use CuDNN kernel. If we set x.use_cudnn=False directly, conv2d will select plain GPU kernel incorrectly. Hence we need to return a new tensor to isolate the action between multiple ops.

Reply:
> 在某些情况下是否可以直接返回输入Tensor以减少开销？

当某个输入 Tensor 只作为单个 op 的输入，不会同时作为多个 op 的输入时，就可以直接返回输入 Tensor。这种判断在框架的角度很难判断，比较依赖于用户的设置

We can return the input tensor directly if this tensor is only used as the input of a single op instead of multiple ops simultaneously. However, it's too hard for the framework to make a such judgment, it strongly depends on the setting of users.

Reply:
> 此函数需要返回 tensor 对象，返回的 tensor 与原 tensor 相比，只有 use_cudnn 属性不一致。
如果返回tensor和原tensor的use_cudnn属性一致是不就可以直接返回输入了？或者这种情况下是否可以直接跳过这个函数的调用？

Reply:
之前做性能分析的时候，Python-C的调度开销还是比较明显的，所以这部分最好尽可能处理的高效一些

Reply:
如果两者的属性一致，直接返回输入。已改正，感谢～

Done, Thx~ 
To optimize the performance of Python-C, if returned and the original tensor has the same `use_cudnn` attribute, return the original tensor directly.